src/main/kotlin/apply/application/EvaluationTargetService.kt

```diff
@@ -140,4 +141,37 @@ class EvaluationTargetService(
             note = request.note
         )
     }
+
+    fun findAllMailSendingTargets(
+        evaluationId: Long,
+    ): List<MailSendingTargetResponse> {
+        val targetApplicantIds = evaluationTargetRepository.findAllByEvaluationId(evaluationId)
+            .map { it.applicantId }
+
+        return applicantRepository.findAllById(targetApplicantIds)
+            .map { MailSendingTargetResponse(it.email) }
+    }
+
+    fun findAllMailSendingTargetsBySpecificEvaluationStatus(
+        evaluationId: Long,
+        evaluationStatus: EvaluationStatus
+    ): List<MailSendingTargetResponse> {
+        val targetApplicantIds = extractMailSendingTargetsByEvaluationStatus(evaluationId, evaluationStatus)
+            .map { it.applicantId }
+
+        return applicantRepository.findAllById(targetApplicantIds)
+            .map { MailSendingTargetResponse(it.email) }
+    }
+
+    private fun extractMailSendingTargetsByEvaluationStatus(
+        evaluationId: Long,
+        evaluationStatus: EvaluationStatus
+    ): List<EvaluationTarget> {
+        if (evaluationStatus == EvaluationStatus.FAIL) {
+            return evaluationTargetRepository.findAllByEvaluationIdAndEvaluationStatus(evaluationId, evaluationStatus)
+                .filter { it.isAllSubmitAssignment() }
+        }
+
+        return evaluationTargetRepository.findAllByEvaluationIdAndEvaluationStatus(evaluationId, evaluationStatus)
```

Comment:
if문보다 if식을 사용하면 어떨까요?

```suggestion
        return if (evaluationStatus == EvaluationStatus.FAIL) {
            evaluationTargetRepository.findAllByEvaluationIdAndEvaluationStatus(evaluationId, evaluationStatus)
                .filter { it.isAllSubmitAssignment() }
        } else {
            evaluationTargetRepository.findAllByEvaluationIdAndEvaluationStatus(evaluationId, evaluationStatus)
        }
```